src/data/complex/R_or_C.lean

```diff
@@ -0,0 +1,603 @@
+/-
+Copyright (c) 2020 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import analysis.normed_space.basic
+import analysis.complex.basic
+
+/-!
+# R_or_C: a typeclass for ℝ or ℂ
+
+This file defines the typeclass is_R_or_C intended to have only two instances:
+ℝ and ℂ. It is meant for definitions and theorems which hold for both the
+real and the complex case, such as inner products and Hilbert spaces. Its API
+follows closely that of ℂ.
+-/
+
+set_option default_priority 100 -- see Note [default priority]
+
+/--
+This typeclass captures properties shared by ℝ and ℂ, with an API that closely matches that of ℂ.
+-/
+class is_R_or_C (K : Type*) [normed_field K] [algebra ℝ K] :=
+(re : K →+ ℝ)
+(im : K →+ ℝ)
+(conj : K →+* K)
+(I : K)                 -- Meant to be set to 0 for K=ℝ
+(of_real : ℝ → K)      -- Meant to be id for K=ℝ and the coercion from ℝ for K=ℂ
+(I_re_ax : re I = 0)
+(I_mul_I_ax : I = 0 ∨ I * I = -1)
+(re_add_im_ax : ∀ (z : K), of_real (re z) + of_real (im z) * I = z)
+(smul_coe_mul_ax : ∀ (z : K) (r : ℝ), r • z = of_real r * z)
+(of_real_re_ax : ∀ r : ℝ, re (of_real r) = r)
+(of_real_im_ax : ∀ r : ℝ, im (of_real r) = 0)
+(mul_re_ax : ∀ z w : K, re (z * w) = re z * re w - im z * im w)
+(mul_im_ax : ∀ z w : K, im (z * w) = re z * im w + im z * re w)
+(conj_re_ax : ∀ z : K, re (conj z) = re z)
+(conj_im_ax : ∀ z : K, im (conj z) = -(im z))
+(conj_I_ax : conj I = -I)
+(norm_sq_eq_def_ax : ∀ (z : K), ∥z∥^2 = (re z) * (re z) + (im z) * (im z))
+(mul_im_I_ax : ∀ (z : K), (im z) * im I = im z)
+(inv_def_ax : ∀ (z : K), z⁻¹ = conj z * of_real ((∥z∥^2)⁻¹))
+(div_I_ax : ∀ (z : K), z / I = -(z * I))
+
+namespace is_R_or_C
+
+variables {K : Type*} [normed_field K] [algebra ℝ K] [is_R_or_C K]
+local notation `𝓚` := @is_R_or_C.of_real K _ _ _
+
+lemma of_real_alg : ∀ x : ℝ, 𝓚 x = x • (1 : K) :=
+  λ x, by rw [←mul_one (𝓚 x), smul_coe_mul_ax]
```

Comment:
```suggestion
λ x, by rw [←mul_one (𝓚 x), smul_coe_mul_ax]
```
Here and below: we generally don't indent the first lines of proofs / definitions. See [the style guide](https://leanprover-community.github.io/contribute/style.html#structuring-definitions-and-theorems).